yileifeng/ramp4-pcar4
# Improvements on instantiation of dynamic components

Reworked the way components are programmatically created and mounted onto app. In addition, fixed the `$t` issues, accessing registered components/directives issues, and `vue-tippy` issues for dynamic fixtures.

[Demo](http://ramp4-app.azureedge.net/demo/users/yileifeng/enhance-dynamic-instantiation/host/index.html)

**Changes**:

- used new `mount` method in `fixture.ts` as a better way to instantiate dynamic fixtures 
- changes to allow dynamic fixtures to access all app registered properties/components/directives + remove unnecessary imports and props now that this is the case 
- fixed all `$t` instances not found and replaced all instances of `t` and `iApi` with the proper `$t` and `$iApi`
- fixed `vue-tippy` hovertips for dynamic components 
- cleaned up some methods and signatures in `fixture.ts`

**Testing**: 

- ensure each of the dynamic map components (panguard/scrollguard, crosshairs, overview map, north arrow, appbar, navbar) are 100% functional 
- check that tooltips display for all these components
